teacher.rb

```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super(age, name, parent_permission: parent_permission)
```

Comment:
- [ ] As I mentioned in the previous comment; to correctly apply the `Inheritance` approach and avoid unnecessary errors, kindly make sure you update your `super` and `initialize` methods as right now they are wrong(**Constructor should extend parent's constructor by adding `@specialization` and a `parameter` for it**). Your both methods should similar to this example below. For as short remind about **Inheritance** I suggest you refer this **[article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#inheritance)**.
```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```